src/Lib/Application.php

```diff
@@ -5,42 +5,48 @@
 use App\Lib\Http;
 use App\Lib\Session;
 use App\Controller\BaseController;
+use Symfony\Component\HttpFoundation\Request;
 use Symfony\Contracts\Service\ServiceProviderInterface;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
 
 class Application
 {
-    private $controllerFactory;
-
     const DEFAULT_CONTROLLER = 'Employee';
     const DEFAULT_ACTION = 'index';
 
-    public function __construct(ServiceProviderInterface $controllerProvider)
+    private $controllerProvider;
+    private $session;
+
+    public function __construct(ServiceProviderInterface $controllerProvider, SessionInterface $session)
```

Comment:
En PHP 8, on peut faire ça directement :
```suggestion
    public function __construct(
        private readonly ServiceProviderInterface $controllerProvider,
        private readonly SessionInterface $session,
    ) {
    }
```

Reply:
Ici je pense qu'il serait préférable d'injecter App/Lib/Session plutôt, car là finalement on embarque la dépendance d'une dépendance (SessionInterface dont dépend App/Lib/Session) 
Cela ne respecte pas vraiment les principes SOLID

Reply:
J'ai horreur de la syntaxe PHP 8 pour constructeur ! Mais bon faut vivre avec son temps :p

Reply:
C'est fait :)